domehunter/dome_control.py

```diff
@@ -355,81 +386,176 @@ def calibrate_dome_encoder_counts(self, num_cal_rotations=2):
-
-        self._stop_moving()
-
-        # set the azimuth per encoder tick factor based on how many ticks we
-        # counted over n rotations
-        self._degrees_per_tick = Angle(
-            360 / (self.encoder_count / rotation_count) * u.deg)
-        # update the az_position_tolerance
-        self.az_position_tolerance = max(
-            self.az_position_tolerance, 2 * self.degrees_per_tick)
-        self.calibrating = False
+        logger.debug(f'calibrate_dome_encoder_counts: Starting calibration rotations.')
+        self._move_event.set()
+        self._rotate_dome(Direction.CW)
+        self._calibrating = True
+        calibrate = threading.Thread(target=self._calibrate)
+        cal_monitor = threading.Thread(target=self._thread_condition,
+                                       args=(self._calibration_completed,))
+        calibrate.start()
+        cal_monitor.start()
 
     def find_home(self):
         """
         Move Dome to home position.
 
         """
+        if self._move_event.is_set():
+            logger.debug('Movement command in progress')
+            raise Exception("Movement command in progress")
+        # iniate the movement and set the _move_event flag
+        logger.debug(f'find_home: Finding Home')
+        self._move_event.set()
         self._rotate_dome(Direction.CW)
         time.sleep(0.1)
-        self._home_sensor.wait_for_active(timeout=self.wait_timeout)
+        homing = threading.Thread(target=self._thread_condition,
+                                  args=(self._find_home_condition,))
+        homing.start()
         if self.testing:
             # in testing mode need to "fake" the activation of the home pin
-            time.sleep(0.5)
-            self._home_sensor_pin.drive_high()
-
-        self._stop_moving()
+            home_pin_high = threading.Timer(0.5,
+                                            self._home_sensor_pin.drive_high)
+            home_pin_high.start()
 
     def sync(self, az):
+        """
+        Sync encoder count to azimuth position.
+        Parameters
+        ----------
+        az : float
+            Desired dome azimuth position in degrees.
+
+        """
+        logger.debug(f'sync: syncing encoder counts to azimuth [{az}]')
         self._encoder_count = self._az_to_ticks(Longitude(az * u.deg))
         return
 
 ###############################################################################
 # Private Methods
 ###############################################################################
 
+    def _thread_condition(self, trigger_condition):
+        """Condition monitoring thread for movement commands.
+
+        Will return when:
+            - trigger_condition is true
+            - abort event is triggered
+            - thread running time exceeds self.wait_timeout
+
+        """
+        goingtoaz = False
+        if trigger_condition.__name__ == '_goto_az_condition':
+            goingtoaz = True
+        start = time.time()
```

Comment:
When you are interested in elapsed time (as here) it is better practice to use `time.monotonic()` instead of `time.time()`.

https://docs.python.org/3.7/library/time.html#time.monotonic

Reply:
ah ok, will keep that in mind for the future. This was just something I added in for debugging. I'll remove it before the PR is merged.